source/plot/LinePlotData2D.ooc

```diff
@@ -57,13 +69,37 @@ LinePlotData2D: class extends PlotData2D {
 		end := FloatPoint2D new(this legendOffset + fontSize, start y)
 		match (this lineStyle) {
 			case LineStyle Dashed =>
-				result = Shapes line(start, end, this lineWidth, this opacity, this color, FloatPoint2D new((this lineWidth * 5) as Float, (this lineWidth * 5) as Float))
+				result = Shapes line(start, end, this lineWidth, this colorBgra, FloatPoint2D new((this lineWidth * 5) as Float, (this lineWidth * 5) as Float))
 			case LineStyle Dotted =>
-				result = Shapes line(start, end, this lineWidth, this opacity, this color, FloatPoint2D new(this lineWidth as Float, this lineWidth as Float))
+				result = Shapes line(start, end, this lineWidth, this colorBgra, FloatPoint2D new(this lineWidth as Float, this lineWidth as Float))
 			case => // LineStyle Solid
-				result = Shapes line(start, end, this lineWidth, this opacity, this color)
+				result = Shapes line(start, end, this lineWidth, this colorBgra)
+		}
+		result = result & Shapes text(FloatPoint2D new(end x, end y + fontSize / 3), this label, fontSize, this colorBgra)
+		result
+	}
+
+	minValues: override func -> FloatPoint2D {
```

Comment:
The result for empty lists might cause arguments and breaking changes in Kean. I would return positive infinity for minimum of nothing so that comparisons can still work.